app/pages/project/workflow-selection.jsx

```diff
@@ -143,13 +153,13 @@ class WorkflowSelection extends React.Component {
 
   clearInactiveWorkflow(selectedWorkflowID) {
     const { preferences } = this.props;
-    const selectedWorkflow = preferences.preferences ? preferences.preferences.selected_workflow : undefined;
-    const projectSetWorkflow = preferences.settings ? preferences.settings.workflow_id : undefined;
+    const selectedWorkflow = (preferences && preferences.preferences) ? this.sanitiseID(preferences.preferences.selected_workflow) : undefined;
```

Comment:
Should this be called `userSelectedWorkflow` for clarity, and for consistency with the same block earlier in the component?

Reply:
Also, I think it would be better if values were sanitised in one place and the sanitised values stored in the instance for use by other component methods.